tecken/tests/conftest.py

```diff
@@ -76,185 +79,111 @@ def test_something(requestsmock):
-
-    Whilst working on a test, you might want wonder "What would happen"
-    if I let this actually use the Internet to make the call un-mocked.
-    To do that use ``botomock.orig()``. For example::
-
-        def test_something(botomock):
-
-            def my_make_api_call(self, operation_name, api_params):
-                if api_params == something:
-                    ...you know what to do...
-                else:
-                    # Only in test debug mode
-                    result = botomock.orig(self, operation_name, api_params)
-                    print(result)
-                    raise NotImplementedError
+def uploaderuser(django_user_model):
+    """Creates and returns a fake user in the uploaders group."""
+    user = django_user_model.objects.create(
+        username="uploader", email="uploader@example.com"
+    )
+    group = Group.objects.get(name="Uploaders")
+    user.groups.add(group)
+    assert user.has_perm("upload.upload_symbols")
+    return user
 
-    """
 
-    class BotoMock:
-        def __init__(self):
-            self.calls = []
+def clear_s3_storage(self: S3Storage):
+    """Make sure the S3 bucket exists and delete all files under the prefix."""
+    # NOTE(smarnach): This gets patched into S3Storage as a method. I don't want this to exist in
+    # production code, since it should never get called there.
+    client = self._get_client()
+    client.create_bucket(Bucket=self.bucket)
+    response = client.list_objects_v2(Bucket=self.bucket, Prefix=self.prefix)
+    for object in response.get("Contents", []):
+        client.delete_object(Bucket=self.bucket, Key=object["Key"])
 
-        def __call__(self, mock_function):
-            def wrapper(f):
-                def inner(*args, **kwargs):
-                    self.calls.append(args[1:])
-                    return f(*args, **kwargs)
 
-                return inner
+S3Storage.clear = clear_s3_storage
 
-            return mock.patch(
-                "botocore.client.BaseClient._make_api_call", new=wrapper(mock_function)
-            )
 
-        def orig(self, *args, **kwargs):
-            return _orig_make_api_call(*args, **kwargs)
+def clear_gcs_storage(self: GCSStorage):
+    """Make sure the GCS bucket exists and delete all files under the prefix."""
+    # NOTE(smarnach): This gets patched into GCSStorage as a method. I don't want this to exist in
+    # production code, since it should never get called there.
+    client = self._get_client()
+    try:
+        client.create_bucket(self.bucket)
+    except Conflict:
+        # Bucket already exists.
+        pass
+    bucket = self._get_bucket()
+    blobs = bucket.list_blobs(prefix=self.prefix, fields="items(name)")
+    bucket.delete_blobs(list(blobs))
 
-    return BotoMock()
 
+GCSStorage.clear = clear_gcs_storage
 
-class S3Helper:
-    """S3 helper class.
 
-    When used in a context, this will clean up any buckets created.
+@pytest.fixture
+def bucket_name(request):
```

Comment:
So far the tests used the same bucket as the local development environment, and the bucket got deleted when running the tests. I found this quite annoying since it meant that running the tests broke the running local environment for no reason. This was my main motivation for the change. Since I was changing it anyway, I decided to also make the tests more independent by using separate buckets. (The only remaining dependency between tests is now the Redis cache, which gets cleared for each test. This means we can't run the tests in parallel yet, but we are close.)